### PR TITLE
Fix intra-doc links on pub re-exports

### PR DESCRIPTION
**Pull request by jyn514**
Partial fix for https://github.com/rust-lang/rust/issues/76073 - This removes the incorrect error, but doesn't show the documentation anywhere.
r? @GuillaumeGomez

### CODE REVIEW

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
The current code looks good though. Ping me once it's fixed. ;)

**Comment by jyn514** [comment]
@guillaumegomez the tests are failing because this documentation is never displayed. It seems weird that it's silently ignored, I can change the test but are you sure this should be merged as is?

**Comment by GuillaumeGomez** [comment]
If the documentation isn't shown, it's invalid so please this issue first. ;)

**Comment by jyn514** [comment]
We discussed this on Discord: the documentation is _never_ shown for pub re-exports. That's not introduced by this PR, that's an existing behavior. So it's fine for now to just fix the bugs in intra-doc links, but we should have a conversation about whether we want to show docs on re-exports.

**Comment by jyn514** [comment]
I think I can replicate the same bug with `///` style comments on a module instead of `//!`, if so I'll change the test to that. If not, I'll just test that `broken_intra_doc_links` doesn't fire on the existing test.

**Comment by jyn514** [comment]
> I think I can replicate the same bug with /// style comments on a module instead of //!, if so I'll change the test to that

It's not possible to write `///` style comments on a top-level modules 😆 So I just test that no false-positive errors are emitted.

**Comment by jyn514** [comment]
```
warning: spurious network error (1 tries remaining): failed to get 200 response from `https://crates.io/api/v1/crates/thread_local/1.0.1/download`, got 502
error: failed to download from `https://crates.io/api/v1/crates/thread_local/1.0.1/download`
```

File: src/test/rustdoc/intra-link-pub-use.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#![deny(broken_intra_doc_links)]
+
+/// [std::env] [g]
+// FIXME: This can't be tested because rustdoc doesn't show documentation on pub re-exports.
```
Comment (ollie27) [inline src/test/rustdoc/intra-link-pub-use.rs:4]: As of #63048 it does but only when inlining from an external crate.

Comment (jyn514) [reply src/test/rustdoc/intra-link-pub-use.rs:4]: Nice! I'll add a test using that :)

**Comment by jyn514** [comment]
Updated!

**Comment by jyn514** [comment]
ping @GuillaumeGomez - is this waiting on anything?

**Comment by GuillaumeGomez** [comment]
Nope, all good!

@bors: r=ollie27,GuillaumeGomez

**Comment by kennytm** [comment]
@bors retry

```
2020-09-03T08:41:55.3983684Z  Downloading crates ...
2020-09-03T08:41:55.3983911Z error: failed to download from `https://crates.io/api/v1/crates/cloudabi/0.0.3/download`
2020-09-03T08:41:55.3984063Z 
2020-09-03T08:41:55.3984213Z Caused by:
2020-09-03T08:41:55.3984404Z   [55] Failed sending data to the peer (Connection died, tried 5 times before giving up)
2020-09-03T08:41:55.3984763Z ', src/tools/tidy/src/deps.rs:202:20
```

**Comment by kennytm** [comment]
@bors retry

5 hour timeout on `macOS x86_64-apple`.

**Pull request merged**